### ISSUE TITLE
Variable x hides variable defined in ... 

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#808]
Using 2.8.1, it seems that #478 has returned:

````
function foo returns logical ( bar as integer ) forward.

define variable bar as integer.
````
![image](https://user-images.githubusercontent.com/5297464/86969632-63042200-c16e-11ea-9155-3d7ffdefd9b3.png)

**Comment by gquerret** [comment]
Did you upgrade from 2.8.0 to 2.8.1 ? I don't think the issue is in the rule, there are unit tests for that and none of them fail. Due to the preprocessor usage in the code, could you check if '{1}' was changed recently, so the code would be `MAP TO xxx IN yyy` instead of `FORWARD` ? The rule don't report issues when there is a `FORWARD` keyword, but doesn't handle `MAP TO`.
There was also a small modification in the preprocessor code, so it may have an impact here.

**Comment by stefandrissen** [comment]
I did upgrade from 2.8.0 to 2.8.1, but I do not know if this was already an issue or not. I only noticed it because I was looking through some of the rules that were flagging issues. 

This code with map has not changed.

**Comment by gquerret** [comment]
Just to verify, do you provide any parameters to the include file ?

**Comment by stefandrissen** [comment]
usage in this .p:
````
{ general/modu/forward/errqueue.i   default }
````
definition at beginning of include:
````
&IF "{1}" = "default" &THEN
   DEFINE VARIABLE p_hperrqueue  AS HANDLE      NO-UNDO.
   &SCOPED-DEFINE HANDLE p_hperrqueue
&ELSEIF "{1}" > "" &THEN
   &SCOPED-DEFINE HANDLE {1}
&ENDIF
````

**Comment by gquerret** [comment]
Unless I'm wrong, preprocessor will generate:
```
function errqueueSetInsertCompany returns logical ( ... ) map to errqueueSetInsertCompany in p_hperrqueue.
```
and not 
```
function errqueueSetInsertCompany returns logical ( ... ) forward.
```

So we end up in a case which is not handled by the rule ; that will be fixed.

**Comment by stefandrissen** [comment]
Yes, the map to allows multiple instances with different names to be used from caller by specifying {2}.

**Comment by gquerret** [comment]
Fix included in 2.8.2 (released a few minutes ago).
